app/controllers/application_controller.rb

```diff
@@ -1,6 +1,8 @@
 class ApplicationController < ActionController::Base
   include UserSessionContext
 
+  FLASH_KEYS = %w[alert error notice success warning].freeze
```

Comment:
What do you think about using the `add_flash_types` Rails convention instead? http://api.rubyonrails.org/classes/ActionController/Flash/ClassMethods.html

That way, we won't need to slice this array in `shared/_flashes`.